AFQ/bundles.py

```diff
@@ -0,0 +1,338 @@
+        sls = dtu.transform_tracking_output(sft.streamlines, affine)
+        return StatefulTractogram(sls,
+                                  reference,
+                                  self.space,
+                                  origin=origin,
+                                  data_per_streamline=sft.data_per_streamline)
+
+    def apply_affine(self, affine, reference, origin=Origin.NIFTI):
+        """
+        Apply a linear transformation, given by affine, to all
+        streamlines.
+
+        Parameters
+        ----------
+        affine : array (4, 4)
+            Apply affine matrix to all streamlines
+
+        reference : Nifti or Trk filename, Nifti1Image or TrkFile,
+            Nifti1Header, trk.header (dict) or another Stateful Tractogram
+            Reference that provides the new spatial attribute.
+
+        origin : Enum (dipy.io.stateful_tractogram.Origin), optional
+            New origin of streamlines.
+            Default: Origin.NIFTI
+        """
+        for bundle_idx, bundle in self.bundles.items():
+            self.bundles[bundle_idx] = self._apply_affine_sft(bundle,
+                                                              affine,
+                                                              reference,
+                                                              origin)
+            logging.disable(level=logging.WARNING)
+        logging.disable(logging.NOTSET)
+
+    def to_space(self, space):
+        """
+        Transform streamlines to space.
+
+        Parameters
+        ----------
+        space : Space
+            Space to transform the streamlines to.
+        """
+        for bundle_idx, _ in self.bundles.items():
+            self.bundles[bundle_idx].to_space(space)
+            logging.disable(level=logging.WARNING)
+        logging.disable(logging.NOTSET)
+
+    def save_bundles(self, file_path='./', file_suffix='.trk',
+                     space=None, bbox_valid_check=False):
+        """
+        Save tractograms in bundles.
+
+        Parameters
+        ----------
+        file_path : string, optional.
+            Path to save trk files to.
+            Default: './'
+        file_suffix : string, optional.
+            File name will be the bundle name + file_suffix.
+            Default: '.trk'
+        space : string
+            Space to save the streamlines in. If not none, the streamlines
+            will be transformed to this space, saved, then transformed back.
+            Default: None.
+        bbox_valid_check : boolean, optional.
+            Whether to verify that the bounding box is valid in voxel space.
+            Default: False
+        """
+        if space is not None:
+            space_temp = self.space
+            self.to_space(space)
+
+        for bundle_name, bundle in self.bundles.items():
+            save_tractogram(bundle,
+                            os.path.join(file_path, bundle_name, file_suffix),
```

Comment:
This wouldn't work, because the result would be something like: 

`'./path/to/bundle_name/.trk'`

```suggestion
                            os.path.join(file_path, f"{bundle_name}{file_suffix}"),
```